packages/react-native/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm

```diff
@@ -194,6 +194,11 @@ - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &
     _backedTextInputView.editable = newTextInputProps.traits.editable;
   }
 
+  if (newTextInputProps.multiline && newTextInputProps.traits.dataDetectorTypes != oldTextInputProps.traits.dataDetectorTypes
```

Comment:
why does it have to be multiline?

Reply:
Since `UITextField` does not support `DataDetectorTypes`, current RN docs also points out that it only works when `multiline={true}` and `editable={false}`

[RN TextInput](https://reactnative.dev/docs/textinput#datadetectortypes-ios)
[UITextView-DataDetectorTypes](https://developer.apple.com/documentation/uikit/uitextview#Formatting-special-data-in-text)
[UITextField](https://developer.apple.com/documentation/uikit/uitextfield)